bucket/coverpoint.py

```diff
@@ -79,12 +86,15 @@ def __init__(self, name: str, description: str):
         # Instance of Bucket class to increment hit count for a bucket
         self.bucket = Bucket(self)
 
-        self._setup()
-
         self.tier = 0
         self.tier_active = True
         self.tags = []
 
+        self._setup()
+        self.name = name if name is not None else self.NAME
+        self.description = description if description is not None else self.DESCRIPTION
+        self.motivation = motivation if motivation is not None else self.MOTIVATION
+
```

Comment:
I don't think these should ever end up as None - not setting them should result in an error, a sensible default, or an empty string